src/services/completions.ts

```diff
@@ -371,6 +381,8 @@ namespace ts.Completions {
                 }
             }
 
+            result = result.filter(entry => entry.name.indexOf(baseName) >= 0);

```

Comment:
Are there any editors that prefer to show all possible completions to provide context, that this would then restrict?

Could we do this check in advance against `foundFileName` at line 363 and `directoryName` at line 380, rather than allocating a number of completion objects that are then thrown out?